src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -433,7 +433,13 @@ namespace winrt::TerminalApp::implementation
                         THROW_IF_FAILED(dialog->SetDefaultExtension(L"txt"));
 
                         // Default to using the tab title as the file name
-                        THROW_IF_FAILED(dialog->SetFileName((control.Title() + L".txt").c_str()));
+                        auto filename = { control.Title() };

```

Comment:
`Title()` returns a `winrt::hstring` which is an immutable string. You first have to make it mutable by copying it into another container like `std::wstring`:
```cpp
std::wstring filename{ control.Title() };
```